lib/charms/opensearch/v0/helper_conf_setter.py

```diff
@@ -272,14 +272,13 @@ def replace(
             output_file: Target file for the result config, by default same as config_file
         """
         path = f"{self.base_path}{config_file}"
-
         if not exists(path):
             raise FileNotFoundError(f"{path} not found.")
 
         with open(path, "r+") as f:
             data = f.read()
 
-            if regex and old_val and re.compile(old_val).match(data):
+            if regex and old_val and re.compile(old_val, re.MULTILINE).findall(data):
```

Comment:
Extended the regex, as it was failing to match: `-Xmx1g` in a file with content: `-Xms1g\n-Xmx1g`. Setting the multiline and forcing it to be more greedy and search for any combinations resolves this problem.